kwave/options/simulation_execution_options.py

```diff
@@ -1,81 +1,143 @@
-    # Whether to delete the input and output files after the simulation
     delete_data: bool = True
-    # GPU device flag
     device_num: Optional[int] = None
-    # number of threads
     num_threads: Union[int, str] = "all"
-
-    # user defined thread binding option
     thread_binding: Optional[bool] = None
-
-    # user input for system string
     system_call: Optional[str] = None
     verbose_level: int = 0
-
-    # determine whether chunking is handled automatically (the default), or manually
     auto_chunking: Optional[bool] = True
-
-    # show simulation log
     show_sim_log: bool = True
 
     def __post_init__(self):
-        self.validate()
+        if self.is_gpu_simulation:
+            warnings.warn(
+                f"Constructor argument `is_gpu_simulation` is deprecated and will be removed in version {DEPRECATION_VERSION}. Use configure() method instead.",
+                DeprecationWarning,
+            )
+        if self.binary_name:
+            warnings.warn(
+                f"Constructor argument `binary_name` is deprecated and will be removed in version {DEPRECATION_VERSION}. Use configure() method instead.",
+                DeprecationWarning,
+            )
+        self.configure(is_gpu_simulation=self.is_gpu_simulation, binary_path=self.binary_path)
 
-        if self.binary_name is None:
-            if self.is_gpu_simulation:
-                self.binary_name = "kspaceFirstOrder-CUDA"
-            else:
-                self.binary_name = "kspaceFirstOrder-OMP"
+    @property
+    def is_gpu_simulation(self):
+        return self._is_gpu_simulation
+
+    @is_gpu_simulation.setter
+    def is_gpu_simulation(self, value):
+        warnings.warn(
+            f"Setting is_gpu_simulation directly is deprecated and will be removed in version {DEPRECATION_VERSION}. Use configure() method instead.",
+            DeprecationWarning,
+        )
+        self._is_gpu_simulation = value
+
+    @property
+    def binary_name(self):
+        return self._binary_name
+
+    @binary_name.setter
+    def binary_name(self, value):
+        warnings.warn(
+            f"Setting binary_name directly is deprecated and will be removed in version {DEPRECATION_VERSION}. Use configure() method instead.",
+            DeprecationWarning,
+        )
+        self._binary_name = value
 
-        if PLATFORM == "windows":
-            if not self.binary_name.endswith(".exe"):
-                self.binary_name += ".exe"
+    @property
+    def binary_path(self):
+        return self._binary_path
+
+    @binary_path.setter
+    def binary_path(self, value):
+        warnings.warn(
+            f"Setting binary_path directly is deprecated and will be removed in version {DEPRECATION_VERSION}. Use configure() method instead.",
+            DeprecationWarning,
+        )
+        self._binary_path = Path(value)
+
+    def configure(self, is_gpu_simulation: bool, binary_path: Optional[str] = None):
```

Comment:
`binary_path` gives an impression that this is path to the executable. However, this is path to the directory with executable. Perhaps update to `binary_dir`?